python/pyarrow/types.pxi

```diff
@@ -207,6 +211,19 @@ cdef class StructType(DataType):
 
     cdef void init(self, const shared_ptr[CDataType]& type):
         DataType.init(self, type)
+        self.struct_type = <const CStructType*> type.get()
+
+    cdef Field child_by_name(self, name):
+        """
+        Access a child field by its name rather than the column index.
+        """
+        cdef shared_ptr[CField] field
+
+        field = self.struct_type.GetChildByName(tobytes(name))
```

Comment:
I think we should rename this to `GetFieldByName` and `field` in `arrow::StructType`. Doesn't need to be in this patch